src/react/Orb.js

```diff
@@ -0,0 +1,305 @@
+      this.columnFields.splice(columnIndex,1);
+     }
+     if (dataIndex > -1){
+      this.dataFields.splice(dataIndex,1);
+     }
+    }
+    
+    break;
+
+   case "column":
+    if (this.columnFields.indexOf(this.newField()[h] .__prop_addField) === -1){
+     this.columnFields.push(this.newField()[h] .__prop_addField);
+
+     if (rowIndex > -1){
+      this.rowFields.splice(columnIndex,1);
+     }
+     if (dataIndex > -1){
+      this.dataFields.splice(dataIndex,1);
+     }
+
+    }
+    break;
+
+   case "data":
+     if (this.dataFields.indexOf(this.newField()[h] .__prop_addField) === -1){
+      this.dataFields.push(this.newField()[h] .__prop_addField);
+
+      if (rowIndex > -1){
+       this.rowFields.splice(columnIndex,1);
+      }
+      if (columnIndex > -1){
+       this.columnFields.splice(dataIndex,1);
+      }
+      
+     }
+     break;
+   }
+  }
+ }
+
+
+
+ // for (var m=0; m<this.newField().length; m++){
+      
+ //  for (var n=0;n<this.orbFields.length;n++){
+    
+ //   if (this.orbFields[n].caption === this.newField()[m].__prop_addField){
+ //    var ft = this.newField()[m].__prop_formatFunction;
+
+ //    this.orbFields[n].dataSettings={
+ //     aggregateFunc:this.newField()[m].__prop_aggregateFunc,
+ //     formatFunc:function(value){
+ //      return d3.format(ft)(value);
+ //      }      
+ //    };
+
+
+ //   }
+    
+ //  }
+ // }
+
+// var context = this;
+function createFormatFunction(ft) {
+     return function (value) {
+         return d3.format(ft)(value);
+     };
+}
+for (var m = 0; m < this.newField().length; m++) {
+      
+  for (var n=0;n<this.orbFields.length;n++){
+    
+   if (this.orbFields[n].caption === this.newField()[m].__prop_addField){
+    var ft = this.newField()[m].__prop_formatFunction;
+ 
+    this.orbFields[n].dataSettings={
+     aggregateFunc:this.newField()[m].__prop_aggregateFunc,
+     formatFunc: createFormatFunction(ft)
+    };
+ 
+ 
+   }
+    
+  }
+}
+
+     
+ if (this.removeField()){
+  for (var j=0;j<this.orbFields.length;j++){
+   if (this.orbFields[j].caption === this.removeField()){
+    this.orbFields.splice(j,1);
+    break;
+
+   }
+  }
+
+  for (var l=0;l<this.rowFields.length;l++){
+   if (this.rowFields[l] === this.removeField()){
+    this.rowFields.splice(l,1);
+    break;
+   }
+  }
+
+  for (var r=0;r<this.columnFields.length;r++){
+   if (this.columnFields[r] === this.removeField()){
+    this.columnFields.splice(r,1);
+    break;
+   }
+  }
+
+  for (var s=0;s<this.dataFields.length;s++){
+   if (this.dataFields[s] === this.removeField()){
+    this.dataFields.splice(s,1);
+    break;
+   }
+  }
+ }
+ 
+
+ var react = React;
+ react.unmountComponentAtNode(document.getElementById(this.id() + "_orb"));
+ this._div = element.append("div").attr("id", this.id() + "_orb");
+ this._orb = new orb.pgridwidget(this.orbConfig(ds,this.orbFields,this.rowFields,this.columnFields,this.dataFields));
+ this._orb.render(document.getElementById(this.id() + "_orb"));
```

Comment:
I agree. Tear down and re-creation is kinda expensive. But the orb.js was written in react.js. If we don't want to complete re-instantiation, IOW if we just want to recreate modified subjects in config object, we have to jump into react.js code. I looked up the html structure. It is pretty complicated there.